featurebyte/session/spark.py

```diff
@@ -297,13 +305,15 @@ def _read_batch(self, cursor: Cursor, schema: Schema, batch_size: int = 1000) ->
         pa.RecordBatch
             None if no more rows are available
         """
+        schema = pa.schema(
+            {metadata[0]: self._get_pyarrow_type(metadata[1]) for metadata in cursor.description}
+        )
```

Comment:
Does the schema change between batches hence the need to reconstruct schema on every batch of data?

Reply:
hm.. no i don't think it does. i moved this in since it looked duplicated between both callers that were hitting it, but i guess i'll move it back out since it is a bit unnecessary to repeat this work multiple times.